feat: child redact option (#1059)
pinojs/pino: docs/api.md

If the `mixin` feature is being used merely to add static metadata to each log message,
then a [child logger ⇗](/docs/child-loggers.md) should be used instead.

> <a id=opt-redact></a>

Reviewer: ```suggestion
<a id="opt-redact"></a>
```

* See [`serializers` option](#opt-serializers)
* See [pino.stdSerializers](#pino-stdSerializers)

#### `options` (Object)

> Custom options for child logger, it will override the parent options if applied.

Reviewer: ```suggestion
Options for child logger. These options will override the parent logger options.
```

##### `options.redact` (Array | Object)

> Setting `options.redact` to array or object will override the parent `redact` options. If
> you want to remove `redact` inherited from parent logger, you should pass `[]`.

Reviewer: ```suggestion
Setting `options.redact` to an array or object will override the parent `redact` options. To remove `redact` options inherited from the parent logger set this value as an empty array (`[]`).
```
